### PR TITLE
Make sure, when plugins cause a GET request to fail, there is always a response body.

### PR DESCRIPTION
**Pull request by ywkaras**
Adds Au test with plugin forcing various errors on various hooks, with default body
or set body.

### CODE REVIEW

**Comment by ywkaras** [comment]
[approve ci Ubuntu]

**Comment by ywkaras** [comment]
[approve ci ubuntu]

**Comment by ywkaras** [comment]
[approve ci clang-analyzer]

**Comment by ywkaras** [comment]
[approve ci autest]

**Comment by ywkaras** [comment]
As alternative to HttpMsgComp.h, Alan proposes https://github.com/SolidWallOfCode/txn_box/blob/master/plugin/include/txn_box/ts_util.h / https://github.com/SolidWallOfCode/txn_box/blob/master/plugin/src/ts_util.cc

**Comment by ywkaras** [comment]
[approve ci centos]

**Comment by bneradt** [comment]
[approve ci ubuntu]

**Comment by bneradt** [comment]
[approve ci autest]

**Comment by ezelkow1** [comment]
[approve ci rocky]

File: plugins/xdebug/xdebug.cc [hunk 1]
```diff
@@ -31,11 +31,11 @@
...
-#include "ts/experimental.h"
-#include "tscore/ink_defs.h"
-#include "tscpp/util/PostScript.h"
-#include "tscpp/util/TextView.h"
-#include "Cleanup.h"
+#include <ts/experimental.h>
+#include <tscore/ink_defs.h>
+#include <tscpp/util/PostScript.h>
+#include <tscpp/util/TextView.h>
+#include <tscpp/api/Cleanup.h>
```
Comment (ywkaras) [inline plugins/xdebug/xdebug.cc:38]: No real change to xdebug, just from this header to make it available to all plugins.

File: proxy/http/HttpTransact.cc [hunk 2]
```diff
@@ -1743,8 +1743,7 @@ HttpTransact::HandleApiErrorJump(State *s)
   **/
   if (s->http_return_code && s->http_return_code >= HTTP_STATUS_BAD_REQUEST) {
     const char *reason = http_hdr_reason_lookup(s->http_return_code);
-    ;
-    build_response(s, &s->hdr_info.client_response, s->client_info.http_version, s->http_return_code, reason ? reason : "Error");
+    build_error_response(s, s->http_return_code, reason ? reason : "Error", nullptr);
```
Comment (ywkaras) [inline proxy/http/HttpTransact.cc:1746]: This is the only real code change.  Most of this PR is to add an Au test.

**Comment by ywkaras** [comment]
Is this an incompatible change?  Since plugin which may have generated empty response bodies in some cases will not generate non-empty response bodies?

**Comment by ywkaras** [comment]
[approve ci autest]

**Comment by ezelkow1** [comment]
[approve ci autest]

**Comment by ywkaras** [comment]
This was an ask from YCPI Prod.  But, no point in fixing it if it's going to continue to rot in the Sun.

**Pull request closed without merging**